extensions/sockets/src/main/java/org/mule/module/socket/api/client/TcpListenerClient.java

```diff
@@ -0,0 +1,183 @@
+import org.mule.module.socket.internal.ConnectionEvent;
+import org.mule.module.socket.internal.TcpInputStream;
+import org.mule.runtime.api.connection.ConnectionException;
+
+import java.io.BufferedInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TcpListenerClient extends AbstractTcpClient implements ListenerSocket
+{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TcpListenerClient.class);
+
+    private ServerSocket socket;
+    private TcpServerSocketProperties socketProperties;
+
+
+    public TcpListenerClient(TcpServerSocketProperties socketProperties, TcpProtocol protocol, String host, Integer port) throws ConnectionException
+    {
+        super(protocol, host, port);
+        this.socketProperties = socketProperties;
+        configureSocket();
+    }
+
+    public void configureSocket() throws ConnectionException
+    {
+        try
+        {
+            this.socket = new ServerSocket();
+
+            if (socketProperties.getTimeout() != null)
+            {
+                socket.setSoTimeout(socketProperties.getTimeout());
+            }
+
+            if (socketProperties.getReceiveBufferSize() != null)
+            {
+
+                socket.setReceiveBufferSize(socketProperties.getReceiveBufferSize());
+            }
+
+            if (socketProperties.getReuseAddress() != null)
+            {
+
+                socket.setReuseAddress(socketProperties.getReuseAddress());
+            }
+        }
+        catch (SocketException e)
+        {
+            throw new ConnectionException("Could not configure socket");
+        }
+        catch (IOException e)
+        {
+            throw new ConnectionException("Could not create socket");
+        }
+
+        try
+        {
+            socket.bind(new InetSocketAddress(host, port), socketProperties.getReceiveBacklog());
+        }
+        catch (IOException e)
+        {
+            throw new ConnectionException(String.format("Could not bind socket to host '%s' and port '%d'", host, port));
+        }
+    }
+
+    private Socket listen() throws ConnectionException
+    {
+        try
+        {
+            // blocking
+            Socket newConnection = socket.accept();
+            configureIncomingConnection(newConnection);
+            return newConnection;
+        }
+        catch (SocketTimeoutException e)
+        {
+            // do nothing because timeout is configurable
+            LOGGER.debug("Socket timed out while listening for a new connection");
+            return null;
+        }
+        catch (IOException e)
+        {
+            // this is the expected behaviour while closing the socket
+            return null;
```

Comment:
according to the accept() javadoc, an IOException could happen if the connection is lost or stale. You should throw a COnnectionException here